.github/workflows/build-and-run-tests-posix.yml

```
@@ -0,0 +1,65 @@
+name: Build and run tests on Linux and MacOS platforms
+
+on:
+  workflow_call:
+    inputs:
+      platform:
+        required: true
+        type: string
+
+jobs:
+  build-and-run-tests-posix:
+    runs-on: ${{ inputs.platform }}
+    steps:
+    - run: echo "Job triggered by ${{ github.event_name }} event, running on ${{ runner.os }}."
+    - run: echo "Branch ${{ github.ref }} on repository ${{ github.repository }}"
+
+    - name: Check out repository code
+      uses: actions/checkout@v3
+
+    - name: Install Ninja
+      run: |
+        if [[ "${{ inputs.platform }}" == ubuntu-latest ]]; then
+          sudo apt-get install ninja-build
+        elif [[ "${{ inputs.platform }}" == macos-latest ]]; then
+          brew install ninja
+        else
+          echo Do not know how to install Ninja on platform ${{ inputs.platform }}
+          false
+        fi
+      shell: bash
+
+    - name: Retrieve Meson from cache
+      id: cache-meson
+      uses: actions/cache@v3
+      with:
+        path: meson-0.63.0
+        key: meson-0.63.0
+
+    - name: Download Meson
+      if: steps.cache-meson.outputs.cache-hit != 'true'
+      run: wget -q https://github.com/mesonbuild/meson/releases/download/0.63.0/meson-0.63.0.tar.gz &&
+           tar -xzf meson-0.63.0.tar.gz &&
+           rm meson-0.63.0.tar.gz
+
+    - name: Create symbolic link for Meson
+      run: ln -s meson-0.63.0/meson.py . &&
+           ls -la
```

I do not really get the reason for the `&&` operator you use here everywhere?
Makes it harder to read, at least for me ;)

(Github suggestions ... whatever, of course this should suggest a 2 line change)
```suggestion
        run: |
            ln -s meson-0.63.0/meson.py .
            ls -la
```

---

If  the first line has `$? -ne 0` it will abort anyhow, and not go to the next line...

---

Fair enough, will edit this a bit...

---

Now we use multi-line scripts instead of the && operator.